lnwallet/btcwallet/btcwallet.go

```diff
@@ -343,6 +366,23 @@ func (b *BtcWallet) Start() error {
 			}
 		}
 
+		// If this is the first startup with remote signing and wallet
+		// migration turned on and the wallet wasn't previously
+		// migrated, we can do that now that we made sure all accounts
+		// that we need were derived correctly.
+		if !walletIsWatchOnly && b.cfg.WatchOnly &&
+			b.cfg.MigrateWatchOnly {
+
+			log.Infof("Migrating wallet to watch-only mode, " +
+				"purging all private key material")
+
+			ns := tx.ReadWriteBucket(waddrmgrNamespaceKey)
+			err = b.wallet.Manager.ConvertToWatchingOnly(ns)
```

Comment:
ok so previously we don't mark the wallet as watch-only in db?

Reply:
Previously a new wallet would be created by `lncli createwatchonly` which set this flag at wallet creation time. If we migrate a wallet that previously had private keys, then this call will set the `watch-only` boolean in the DB, yes.

Reply:
Got it. With this commit I guess we can now turn a non-watch-only node into a watch-only node.